src/app/settings/admin/constraint-templates/component.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnChanges, OnDestroy, OnInit, ViewChild} from '@angular/core';
+import {MatDialog, MatDialogConfig} from '@angular/material/dialog';
+import {MatPaginator} from '@angular/material/paginator';
+import {MatSort} from '@angular/material/sort';
+import {MatTableDataSource} from '@angular/material/table';
+import {AppConfigService} from '@app/config.service';
+import {ApiService} from '@core/services/api/service';
+import {NotificationService} from '@core/services/notification/service';
+import {UserService} from '@core/services/user/service';
+import {ConfirmationDialogComponent} from '@shared/components/confirmation-dialog/confirmation-dialog.component';
+import {ConstraintTemplate} from '@shared/entity/opa';
+import * as _ from 'lodash';
+import {Subject, timer} from 'rxjs';
+import {takeUntil, filter, take, switchMap} from 'rxjs/operators';
+import {ConstraintTemplatesDataDialogComponent} from './constraint-templates-data-dialog/component';
+
+@Component({
+  selector: 'km-constraint-templates',
+  templateUrl: './template.html',
+})
+export class ConstraintTemplatesComponent implements OnInit, OnChanges, OnDestroy {
+  constraintTemplates: ConstraintTemplate[] = [];
+  dataSource = new MatTableDataSource<ConstraintTemplate>();
+  displayedColumns: string[] = ['templateName', 'actions'];
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
+  private readonly _refreshTime = 10; // in seconds
+  private _unsubscribe = new Subject<void>();
+
+  constructor(
+    private readonly _apiService: ApiService,
+    private readonly _userService: UserService,
+    private readonly _notificationService: NotificationService,
+    private readonly _matDialog: MatDialog,
+    private readonly _appConfig: AppConfigService
+  ) {}
+
+  ngOnInit() {
+    this.dataSource.data = this.constraintTemplates;
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+    this.sort.active = 'templateName';
+    this.sort.direction = 'asc';
+
+    timer(0, this._refreshTime * this._appConfig.getRefreshTimeBase())
```

Comment:
This part introduces a significant memory leak to the application. It will create a new subscription every X seconds without getting rid of the old ones.

Reply:
i will have a look

Reply:
Is this solved?

Reply:
I think yes, I was trying to implement it within the latest commit. So it depends on your judgment 😅

Reply:
It's not. It still creates subscription in subscription every 10 seconds without closing the previous one.

Reply:
I've removed this part in https://github.com/kubermatic/dashboard/pull/2999/commits/fef7383923ae5a760bba51435cf876320ef57d9e 🤔 Or am I getting something wrong?

Reply:
oh... github didn't refresh for me and I was still seeing the old code. It's partially better. I'd improve further though. Now it would refresh only in case user triggered some action. There is no autorefresh similar to what we do for most resources. You could do it similar to datacenters and add refresh + caching inside the service directly.

Reply:
like https://github.com/kubermatic/dashboard/pull/2999/commits/cd4c6044c962747ab59bc867065b855246bc3479 ?